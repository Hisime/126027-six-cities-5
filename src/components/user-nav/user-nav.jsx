import React from "react";
import {UserNavAuth} from '../user-nav-auth/user-nav-auth';
import {UserNavUnauth} from "../user-nav-unauth/user-nav-unauth";
import {connect} from "react-redux";
import {Link} from "react-router-dom";
import {AppRoute} from "../../consts";
import {getUser} from "../../store/selectors/selectors";
import {userPropType} from "../../propTypes";

const UserNav = (props) => {
  const {user} = props;
  const getComponentByType = () => {
    if (user) {
      return <UserNavAuth user={user}/>;
    }
    return <UserNavUnauth/>;
  };
  return (
    <li className="header__nav-item user">
      <Link className="header__nav-link header__nav-link--profile" to={AppRoute.FAVORITES}>
        {getComponentByType()}
      </Link>
    </li>
  );
};

UserNav.propTypes = {
  user: userPropType,
};

const mapStateToProps = (state) => ({
  user: getUser(state),
});

export {UserNav};
export default connect(mapStateToProps)(UserNav);

