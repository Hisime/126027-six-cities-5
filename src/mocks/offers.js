import {OfferType} from '../consts';
import {Features} from '../consts';

export default [
  {
    id: 1,
    pictures: [
      {
        src: `img/apartment-02.jpg`,
        title: `Photo studio`,
      },
      {
        src: `img/room.jpg`,
        title: `Photo studio2`,
      },
      {
        src: `img/apartment-01.jpg`,
        title: `Photo studio3`,
      },
      {
        src: `img/apartment-03.jpg`,
        title: `Photo studio4`,
      },
      {
        src: `img/apartment-01.jpg`,
        title: `Photo studio5`,
      },
      {
        src: `img/studio-01.jpg`,
        title: `Photo studio6`,
      }
    ],
    premium: true,
    title: `Beautiful & luxurious studio at great location`,
    favorite: true,
    rate: 4.8,
    type: OfferType.APARTMENT,
    bedRoomsCount: 3,
    guestsCount: 4,
    price: 120,
    host: {
      image: `img/avatar-angelina.jpg`,
      name: `Angelina`,
      super: true,
      about: `Friendly and positive girl`
    },
    features: [
      Features.WIFI,
      Features.HEATING,
      Features.KITCHEN,
      Features.FRIDGE,
      Features.WASHING_MACHINE,
      Features.COFFEE_MACHINE,
      Features.DISHWASHER,
      Features.TOWELS,
      Features.CABEL_TV
    ],
    reviews: [
      {
        avatar: `img/avatar-max.jpg`,
        name: `Max`,
        rate: 3,
        date: `April 2019`,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`
      },
      {
        avatar: `img/avatar-max.jpg`,
        name: `Edvard`,
        rate: 5,
        date: `April 2019`,
        text: `Perfect place`
      }
    ],
    coords: [52.3909553943508, 4.85309666406198]
  },
  {
    id: 2,
    pictures: [
      {
        src: `img/apartment-02.jpg`,
        title: `Photo studio`,
      },
      {
        src: `img/room.jpg`,
        title: `Photo studio2`,
      },
      {
        src: `img/apartment-01.jpg`,
        title: `Photo studio3`,
      },
      {
        src: `img/apartment-03.jpg`,
        title: `Photo studio4`,
      },
      {
        src: `img/apartment-01.jpg`,
        title: `Photo studio5`,
      },
      {
        src: `img/studio-01.jpg`,
        title: `Photo studio6`,
      }
    ],
    premium: true,
    title: `Small & cozy room`,
    favorite: false,
    rate: 4,
    type: OfferType.ROOM,
    bedRoomsCount: 1,
    guestsCount: 2,
    price: 10,
    host: {
      image: `img/avatar-angelina.jpg`,
      name: `Helena`,
      super: false,
      about: `Just normal host`
    },
    features: [
      Features.WIFI,
      Features.HEATING,
      Features.KITCHEN,
      Features.FRIDGE
    ],
    reviews: [
      {
        avatar: `img/avatar-max.jpg`,
        name: `Leo`,
        rate: 3,
        date: `April 2019`,
        text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`
      },
      {
        avatar: `img/avatar-max.jpg`,
        name: `Edvard`,
        rate: 5,
        date: `April 2019`,
        text: `Perfect place`
      }
    ],
    coords: [52.369553943508, 4.85309666406198]
  },
  {
    id: 3,
    pictures: [
      {
        src: `img/apartment-02.jpg`,
        title: `Photo studio`,
      },
      {
        src: `img/room.jpg`,
        title: `Photo studio2`,
      },
      {
        src: `img/apartment-01.jpg`,
        title: `Photo studio3`,
      },
      {
        src: `img/apartment-03.jpg`,
        title: `Photo studio4`,
      },
      {
        src: `img/apartment-01.jpg`,
        title: `Photo studio5`,
      },
      {
        src: `img/studio-01.jpg`,
        title: `Photo studio6`,
      }
    ],
    premium: true,
    title: `Bright large house, arty, bohemian decor`,
    favorite: false,
    rate: 4.8,
    type: OfferType.HOUSE,
    bedRoomsCount: 2,
    guestsCount: 5,
    price: 1000,
    host: {
      image: `img/avatar-angelina.jpg`,
      name: `Max`,
      super: true,
      about: `I work from home most days, I’m a photographer, so I’m around to help with any queries on the area.`
    },
    features: [
      Features.WIFI,
      Features.HEATING,
      Features.KITCHEN,
      Features.FRIDGE,
      Features.WASHING_MACHINE,
      Features.COFFEE_MACHINE,
      Features.DISHWASHER,
      Features.TOWELS,
      Features.CABEL_TV
    ],
    reviews: [
      {
        avatar: `img/avatar-max.jpg`,
        name: `Peter`,
        rate: 4.7,
        date: `April 2019`,
        text: `Mark is a fab host! The house is perfect location for great local spots`
      },
      {
        avatar: `img/avatar-max.jpg`,
        name: `Edvard`,
        rate: 5,
        date: `April 2019`,
        text: `Mark is a fab host! The house is perfect location for great local spots.`
      }
    ],
    coords: [52.3909553943508, 4.929309666406198]
  },
  {
    id: 4,
    pictures: [
      {
        src: `img/apartment-02.jpg`,
        title: `Photo studio`,
      },
      {
        src: `img/room.jpg`,
        title: `Photo studio2`,
      },
      {
        src: `img/apartment-01.jpg`,
        title: `Photo studio3`,
      },
      {
        src: `img/apartment-03.jpg`,
        title: `Photo studio4`,
      },
      {
        src: `img/apartment-01.jpg`,
        title: `Photo studio5`,
      },
      {
        src: `img/studio-01.jpg`,
        title: `Photo studio6`,
      }
    ],
    premium: false,
    title: `Bright large house, arty, bohemian decor`,
    favorite: true,
    rate: 3,
    type: OfferType.HOTEL,
    bedRoomsCount: 1,
    guestsCount: 2,
    price: 50,
    host: {
      image: `img/avatar-angelina.jpg`,
      name: `Hotel room 5min to Tower Bridge`,
      super: true,
      about: `I am very friendly and communicative`
    },
    features: [
      Features.WIFI,
      Features.HEATING,
      Features.KITCHEN,
      Features.FRIDGE,
      Features.WASHING_MACHINE,
      Features.COFFEE_MACHINE,
      Features.DISHWASHER,
      Features.TOWELS,
    ],
    reviews: [
      {
        avatar: `img/avatar-max.jpg`,
        name: `Dan`,
        rate: 4.9,
        date: `April 2018`,
        text: `Great clean place. Very quiet. And a lovely helpful host.`
      },
      {
        avatar: `img/avatar-max.jpg`,
        name: `Edvard`,
        rate: 5,
        date: `April 2017`,
        text: `Great clean place. Very quiet. And a lovely helpful host.`
      }
    ],
    coords: [52.3809553943508, 4.939309666406198]
  }
];
