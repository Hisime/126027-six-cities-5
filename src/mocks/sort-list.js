import {SortType} from "../consts";

export default [
  {
    value: SortType.POPULAR,
    label: `Popular`,
  },
  {
    value: SortType.PRICE_ASC,
    label: `Price: low to high`,
  },
  {
    value: SortType.PRICE_DES,
    label: `Price: high to low`,
  },
  {
    value: SortType.TOP_DES,
    label: `Top rated first`,
  },
];
