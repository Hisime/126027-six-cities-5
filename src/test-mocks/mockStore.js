import {createStore} from "redux";
import rootReducer from "../store/reducers/root-reducer";

export const mockStore = createStore(rootReducer);
