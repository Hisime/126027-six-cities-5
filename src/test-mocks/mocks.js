/* eslint-disable */

export const user = {
  avatar_url: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg`,
  email: `adawd@awda.ru`,
  id: 1,
  is_pro: false,
  name: `adawd`,
};

export const match = {
  params: {
    id: `1`,
  },
};

export const citiesMock = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`,
];

export const noop = () => {};
