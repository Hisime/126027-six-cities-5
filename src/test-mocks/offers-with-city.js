export const offersWithCity = {
  "Paris":
    [{
      "city": {"name": `Paris`, "location": {"latitude": 48.85661, "longitude": 2.351499, "zoom": 13}},
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
      "images": [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`],
      "title": `The Pondhouse - A Magical Place`,
      "is_favorite": true,
      "is_premium": false,
      "rating": 2.2,
      "type": `house`,
      "bedrooms": 3,
      "max_adults": 3,
      "price": 111,
      "goods": [`Laptop friendly workspace`, `Breakfast`, `Washer`],
      "host": {"id": 25, "name": `Angelina`, "is_pro": true, "avatar_url": `img/avatar-angelina.jpg`},
      "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      "location": {"latitude": 48.861610000000006, "longitude": 2.340499, "zoom": 16},
      "id": 66,
    }, {
      "city": {"name": `Paris`, "location": {"latitude": 48.85661, "longitude": 2.351499, "zoom": 13}},
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      "images": [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`],
      "title": `Waterfront with extraordinary view`,
      "is_favorite": true,
      "is_premium": false,
      "rating": 4.3,
      "type": `hotel`,
      "bedrooms": 5,
      "max_adults": 7,
      "price": 293,
      "goods": [`Laptop friendly workspace`, `Dishwasher`, `Towels`, `Breakfast`, `Coffee machine`, `Washer`, `Fridge`, `Air conditioning`, `Baby seat`],
      "host": {"id": 25, "name": `Angelina`, "is_pro": true, "avatar_url": `img/avatar-angelina.jpg`},
      "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      "location": {"latitude": 48.868610000000004, "longitude": 2.342499, "zoom": 16},
      "id": 46,
    }, {
      "city": {"name": `Paris`, "location": {"latitude": 48.85661, "longitude": 2.351499, "zoom": 13}},
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
      "images": [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`],
      "title": `Tile House`,
      "is_favorite": true,
      "is_premium": false,
      "rating": 2.9,
      "type": `hotel`,
      "bedrooms": 2,
      "max_adults": 5,
      "price": 266,
      "goods": [`Air conditioning`, `Baby seat`, `Towels`, `Washer`, `Breakfast`, `Fridge`, `Laptop friendly workspace`],
      "host": {"id": 25, "name": `Angelina`, "is_pro": true, "avatar_url": `img/avatar-angelina.jpg`},
      "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      "location": {"latitude": 48.865610000000004, "longitude": 2.350499, "zoom": 16},
      "id": 87,
    }, {
      "city": {"name": `Paris`, "location": {"latitude": 48.85661, "longitude": 2.351499, "zoom": 13}},
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
      "images": [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`],
      "title": `Beautiful & luxurious apartment at great location`,
      "is_favorite": true,
      "is_premium": false,
      "rating": 4.2,
      "type": `apartment`,
      "bedrooms": 4,
      "max_adults": 5,
      "price": 119,
      "goods": [`Laptop friendly workspace`, `Air conditioning`, `Breakfast`, `Towels`, `Baby seat`, `Washer`],
      "host": {"id": 25, "name": `Angelina`, "is_pro": true, "avatar_url": `img/avatar-angelina.jpg`},
      "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
      "location": {"latitude": 48.858610000000006, "longitude": 2.330499, "zoom": 16},
      "id": 76,
    }], "Brussels":
    [{
      "city": {"name": `Brussels`, "location": {"latitude": 50.846557, "longitude": 4.351697, "zoom": 13}},
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      "images": [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`],
      "title": `The Joshua Tree House`,
      "is_favorite": true,
      "is_premium": false,
      "rating": 3.8,
      "type": `house`,
      "bedrooms": 2,
      "max_adults": 10,
      "price": 931,
      "goods": [`Fridge`, `Washer`, `Coffee machine`, `Washing machine`, `Breakfast`, `Dishwasher`, `Towels`, `Air conditioning`, `Baby seat`, `Laptop friendly workspace`],
      "host": {"id": 25, "name": `Angelina`, "is_pro": true, "avatar_url": `img/avatar-angelina.jpg`},
      "description": `Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.`,
      "location": {"latitude": 50.860557, "longitude": 4.376697, "zoom": 16},
      "id": 53,
    }, {
      "city": {"name": `Brussels`, "location": {"latitude": 50.846557, "longitude": 4.351697, "zoom": 13}},
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
      "images": [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`],
      "title": `The house among olive `,
      "is_favorite": true,
      "is_premium": false,
      "rating": 4.6,
      "type": `apartment`,
      "bedrooms": 4,
      "max_adults": 7,
      "price": 218,
      "goods": [`Washer`, `Laptop friendly workspace`, `Breakfast`, `Air conditioning`, `Baby seat`],
      "host": {"id": 25, "name": `Angelina`, "is_pro": true, "avatar_url": `img/avatar-angelina.jpg`},
      "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
      "location": {"latitude": 50.865556999999995, "longitude": 4.371696999999999, "zoom": 16},
      "id": 3,
    }, {
      "city": {"name": `Brussels`, "location": {"latitude": 50.846557, "longitude": 4.351697, "zoom": 13}},
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
      "images": [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`],
      "title": `House in countryside`,
      "is_favorite": true,
      "is_premium": true,
      "rating": 2.2,
      "type": `room`,
      "bedrooms": 1,
      "max_adults": 3,
      "price": 232,
      "goods": [`Breakfast`, `Washer`, `Laptop friendly workspace`],
      "host": {"id": 25, "name": `Angelina`, "is_pro": true, "avatar_url": `img/avatar-angelina.jpg`},
      "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
      "location": {"latitude": 50.862556999999995, "longitude": 4.375697, "zoom": 16},
      "id": 14,
    }, {
      "city": {"name": `Brussels`, "location": {"latitude": 50.846557, "longitude": 4.351697, "zoom": 13}},
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
      "images": [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`],
      "title": `Amazing and Extremely Central Flat`,
      "is_favorite": true,
      "is_premium": false,
      "rating": 4.3,
      "type": `apartment`,
      "bedrooms": 3,
      "max_adults": 8,
      "price": 323,
      "goods": [`Towels`, `Baby seat`, `Laptop friendly workspace`, `Dishwasher`, `Washing machine`, `Air conditioning`, `Washer`, `Coffee machine`, `Breakfast`, `Fridge`],
      "host": {"id": 25, "name": `Angelina`, "is_pro": true, "avatar_url": `img/avatar-angelina.jpg`},
      "description": `This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
      "location": {"latitude": 50.839557, "longitude": 4.346697, "zoom": 16},
      "id": 2,
    }], "Dusseldorf":
    [{
      "city": {"name": `Dusseldorf`, "location": {"latitude": 51.225402, "longitude": 6.776314, "zoom": 13}},
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
      "images": [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`],
      "title": `Penthouse, 4-5 rooms + 5 balconies`,
      "is_favorite": true,
      "is_premium": false,
      "rating": 4.2,
      "type": `apartment`,
      "bedrooms": 1,
      "max_adults": 2,
      "price": 125,
      "goods": [`Breakfast`, `Laptop friendly workspace`, `Air conditioning`, `Washer`],
      "host": {"id": 25, "name": `Angelina`, "is_pro": true, "avatar_url": `img/avatar-angelina.jpg`},
      "description": `Design interior in most sympathetic area! Complitely renovated, well-equipped, cosy studio in idyllic, over 100 years old wooden house. Calm street, fast connection to center and airport.`,
      "location": {"latitude": 51.211402, "longitude": 6.756314000000001, "zoom": 16},
      "id": 10,
    }],
};
