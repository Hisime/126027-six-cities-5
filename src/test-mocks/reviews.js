export const reviewsMock = [{
  "id": 1,
  "user": {
    "id": 11,
    "is_pro": false,
    "name": `Jack`,
    "avatar_url": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/2.jpg`,
  },
  "rating": 5,
  "comment": `This villa is perfect in every way: the view on mountains and waterfalls, the hot tub and the villa itself. The evening here became a great continuation of our journeys over country.`,
  "date": `2020-10-10T13:38:44.753Z`,
}, {
  "id": 2,
  "user": {
    "id": 18,
    "is_pro": true,
    "name": `Sophie`,
    "avatar_url": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/9.jpg`,
  },
  "rating": 2,
  "comment": `I stayed here for one night and it was an unpleasant experience.`,
  "date": `2020-10-10T13:38:44.753Z`,
}, {
  "id": 3,
  "user": {
    "id": 19,
    "is_pro": false,
    "name": `Christina`,
    "avatar_url": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/10.jpg`,
  },
  "rating": 4,
  "comment": `I stayed here for one night and it was an unpleasant experience.`,
  "date": `2020-10-10T13:38:44.753Z`,
}];
