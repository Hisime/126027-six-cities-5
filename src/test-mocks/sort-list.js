import {sortTypes} from "../consts";

export const sortListMock = [
  {
    value: sortTypes.POPULAR,
    label: `Popular`,
  },
  {
    value: sortTypes.PRICE_ASC,
    label: `Price: low to high`,
  },
  {
    value: sortTypes.PRICE_DES,
    label: `Price: high to low`,
  },
  {
    value: sortTypes.TOP_DES,
    label: `Top rated first`,
  },
];
