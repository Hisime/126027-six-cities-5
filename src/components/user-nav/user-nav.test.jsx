import React from "react";
import renderer from "react-test-renderer";
import {user} from "../../test-mocks/mocks";
import {UserNav} from "./user-nav";
import {Router} from "react-router-dom";
import browserHistory from "../../browser-history/browser-history";

it(`Should UserNav render correctly`, () => {
  const tree = renderer
    .create(
        <Router history={browserHistory}>
          <UserNav
            user={user}
          />
        </Router>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
