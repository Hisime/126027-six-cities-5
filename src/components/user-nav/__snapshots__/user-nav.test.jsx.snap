// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should UserNav render correctly 1`] = `
<li
  className="header__nav-item user"
>
  <a
    className="header__nav-link header__nav-link--profile"
    href="/favorites"
    onClick={[Function]}
  >
    <div
      className="header__avatar-wrapper user__avatar-wrapper"
      style={
        Object {
          "backgroundImage": "url(https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg)",
        }
      }
    />
    <span
      className="header__user-name user__name"
    >
      adawd@awda.ru
    </span>
  </a>
</li>
`;
